tests/pytorch/iris.py

```diff
@@ -106,5 +106,38 @@ def test_step(self, batch, batch_idx):
         self.log("test_acc", self.test_acc.compute())
 
 
+class IrisClassificationMultiOptimizer(IrisClassificationBase):
+    """
+    Contrived lightning module that uses multiple optimizers. In real-world scenarios
+    multiple optimizers might be used for Generative Adversarial Networks (GANs).
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__()
+        self.automatic_optimization = False
+
+    def training_step(self, batch, batch_idx):
```

Comment:
So with this change, we're looking at for, if we set 20 training steps, the total global step count will be 40, and if we specify to log every 5 steps, we'll get 4 logged entries corresponding to the global steps 10, 20, 30, and 40? 
Sounds like a good fix to me!

Reply:
Yep that's right, although the global steps are actually off by 2 (8, 18, 28 and 38) due to the `+ 1` in the condition. I guess the original intention of that was to avoid logging at step 0 although I don't think this callback method is called when step == 0, but I didn't want to change the existing behaviour in the case of a single optimizer.